### ISSUE TITLE
How to sanitize array

### ISSUE THREAD

**Issue by paulschreiber** [WordPress-Coding-Standards/WordPress-Coding-Standards#458]
I'm getting an non-sanitized input error:

    267 | ERROR | Detected usage of a non-sanitized input variable: $_POST

Here's the code:
```php
$taxonomies = isset( $_POST['announcement']['display'] ) ? 
                          wp_unslash( $_POST['announcement']['display'] ) :
                          array(); // input var okay
```

`$_POST['announcement']['display']` looks like this:
```
array (
  'verticals' => 
  array (
    0 => 'life',
    1 => 'politics',
  ),
)
```

I can't run `array_map( 'sanitize_text_field', $_POST['announcement']['display'] )` as that clobbers the array. I do the sanitizing later:
```php
$announcement['display'] = array();

if ( is_array( $taxonomies ) ) {
	foreach ( $taxonomies as $taxonomy => $taxonomy_terms ) {
		foreach ( $taxonomy_terms as $taxonomy_term ) {
			$announcement['display'][ $taxonomy ][] = sanitize_text_field( $taxonomy_term );
		}
	}
}
```

Is there a way to rewrite this to avoid the warning? If not, how can we improve the sniffer to handle this case?

**Comment by JDGrimes** [comment]
> Is there a way to rewrite this to avoid the warning?

Yes, put the sanitization code into a function, `my_sanitize_announcement_data()`. Then, add that function to the list of sanitizing functions:

```xml
<rule ref="WordPress.VIP.ValidatedSanitizedInput">
		<properties>
			<property name="customSanitizingFunctions" value="my_sanitize_announcement_data" type="array" />
		</properties>
	</rule>
```

**Comment by JDGrimes** [comment]
I've just created [a wiki page](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/wiki/Sanitizing-array-input-data) based on this question.
